docs/device/test_device.md

```diff
@@ -0,0 +1,22 @@
+# Example stream device
+
+**Under Construction:** This section will be populated when devices are released.
+
+## Buffer Structure
+- **Constituents**: The buffer consists of a concatenation of 32-bit header data and 8-bit pixel data.
+- **Storage**: A single image is split and stored in a circulating buffer within the device. The [`num_buffers`](../api/stream_daq.md) should match the number of circulating buffers in the device.
+
+## Header Values and Expected Transitions
+
+- **`preamble`**: 32-bit preamble for detecting the beginning of each buffer. The [`preamble`](../api/stream_daq.md) in the device config needs to match the preamble defined in firmware.
```

Comment:
alternatively, if this is information that's tied to the firmware, i have been using a tool recently for doing arbitrary doctests and i can rig us up something to test for changes and compare them here and there ;)

Reply:
Yes, it's better to move general stuff to the model, so I updated it. I initially put everything here because buffer structure and preambles are lower-layer than the model, and there is some device-specific stuff, like **this parameter is not used**, etc., but I guess I'll just leave those specifics.

Reply:
One related question if you have an idea: do you know a convenient way to document model formats and containers with the same docstring? or even just automatically linking them together.

Reply:
Also, I wonder if all devices we want to cover will be 32-bit, though we can think it's always 32-bit for now.

Reply:
> do you know a convenient way to document model formats and containers with the same docstring?

The model/format thing is a little awkward, i was hoping we could settle these models down a bit eventually and convert them into proper schemas where we could add documentation in a more structured way, but let's defer that for a bit while there's still flux.